pkg/collector/corechecks/oracle/activity.go

```diff
@@ -0,0 +1,211 @@
+	sid,
+	serial#,
+	username,
+    osuser,
+    process, 
+    machine, 
+    program ,
+    type,
+    sql_id,
+	force_matching_signature,
+    sql_plan_hash_value,
+    sql_exec_start,
+    module,
+    action,
+    client_info,
+    logon_time,
+    client_identifier,
+    CASE WHEN blocking_session_status = 'VALID' THEN
+	  blocking_instance
+	ELSE
+	  null
+	END blocking_instance,
+    CASE WHEN blocking_session_status = 'VALID' THEN
+		blocking_session
+	ELSE
+		null
+	END blocking_session,
+	CASE WHEN final_blocking_session_status = 'VALID' THEN
+    	final_blocking_instance
+	ELSE
+		null
+	END final_blocking_instance,
+	CASE WHEN final_blocking_session_status = 'VALID' THEN
+    	final_blocking_session
+	ELSE
+		null
+	END final_blocking_session,
+    CASE WHEN state = 'WAITING' THEN
+		event
+	ELSE
+		'CPU'
+	END event,
+	CASE WHEN state = 'WAITING' THEN
+    	wait_class
+	ELSE
+		'CPU'
+	END wait_class,
+	sql_text,
+	pdb_name
+FROM sys.dd_session
+WHERE 
+	( sql_text NOT LIKE '%DD_ACTIVITY_SAMPLING%' OR sql_text is NULL ) 
+	AND (
+		NOT (state = 'WAITING' AND wait_class = 'Idle')
+		OR state = 'WAITING' AND event = 'fbar timer' AND type = 'USER'
+	)`
+
+type RowMetadata struct {
+	Commands       []string `json:"dd_commands,omitempty"`
+	Tables         []string `json:"dd_tables,omitempty"`
+	Comments       []string `json:"dd_comments,omitempty"`
+	QueryTruncated string   `json:"query_truncated,omitempty"`
+}
+
+type OracleActivityRow struct {
+	SessionID              uint64  `db:"SID" json:"sid,omitempty"`
+	SessionSerial          uint64  `db:"SERIAL#" json:"serial,omitempty"`
+	Username               *string `db:"USERNAME" json:"username,omitempty"`
+	OsUser                 *string `db:"OSUSER" json:"os_user,omitempty"`
+	Process                *string `db:"PROCESS" json:"process,omitempty"`
+	Machine                *string `db:"MACHINE" json:"machine,omitempty"`
+	Program                *string `db:"PROGRAM" json:"program,omitempty"`
+	Type                   *string `db:"TYPE" json:"type,omitempty"`
+	SqlID                  *string `db:"SQL_ID" json:"sql_id,omitempty"`
+	ForceMatchingSignature *uint64 `db:"FORCE_MATCHING_SIGNATURE" json:"force_matching_signature,omitempty"`
+	SqlPlanHashValue       *uint64 `db:"SQL_PLAN_HASH_VALUE" json:"sql_plan_hash_value,omitempty"`
+	SqlExecStart           *string `db:"SQL_EXEC_START" json:"sql_exec_start,omitempty"`
+	Module                 *string `db:"MODULE" json:"module,omitempty"`
+	Action                 *string `db:"ACTION" json:"action,omitempty"`
+	ClientInfo             *string `db:"CLIENT_INFO" json:"client_info,omitempty"`
+	LogonTime              *string `db:"LOGON_TIME" json:"logon_time,omitempty"`
+	ClientIdentifier       *string `db:"CLIENT_IDENTIFIER" json:"client_identifier,omitempty"`
+	BlockingInstance       *uint64 `db:"BLOCKING_INSTANCE" json:"blocking_instance,omitempty"`
+	BlockingSession        *uint64 `db:"BLOCKING_SESSION" json:"blocking_session,omitempty"`
```

Comment:
So it looks like maybe this one deserves to be a pointer since I imagine maybe `0` could be a valid blocking session id that we'd want to distinguish from the absence of a blocking session. Can you confirm my intuition?

Reply:
I've never seen a session with `SID=0`:

`SQL> select min(sid) from v$session ;

  MIN(SID)
----------
	 1`

Since the official documentation, however, isn't explicit about it, it would be better to play it safe and distinguish between `0` and `null`

Reply:
Thanks for checking 🙇 .